### PR TITLE
code optimize use func

### PR DESCRIPTION
**Pull request by zgfzgf**
#4018 
use functions whenever possible

### CODE REVIEW

File: miner/miner.go [hunk 1]
```diff
@@ -288,16 +288,10 @@ minerLoop:
 			//
 			// See:  https://github.com/filecoin-project/lotus/issues/1845
 			nextRound := time.Unix(int64(base.TipSet.MinTimestamp()+build.BlockDelaySecs*uint64(base.NullRounds))+int64(build.PropagationDelaySecs), 0)
-
-			select {
-			case <-build.Clock.After(build.Clock.Until(nextRound)):
```
Comment (whyrusleeping) [inline miner/miner.go:293]: I don't think this is right, we explicitly need to handle the stop case here

**Comment by zgfzgf** [comment]
stop case use func 

https://github.com/filecoin-project/lotus/blob/4b61f1756e416f6045a6292ab9418e0210b9b863/miner/miner.go#L155-L160
For example code update：
 	stopping := m.stopping 
 	m.stop = nil 
       **add code**
 	m.stopping = nil 
 	close(stopping) 
 	return 

**here fail**
https://github.com/filecoin-project/lotus/blob/4b61f1756e416f6045a6292ab9418e0210b9b863/miner/miner.go#L294-L299

File: miner/miner.go (hunk 1 above)
Comment (magik6k) [reply miner/miner.go:293]: We also handle stopping at the top of this loop

File: miner/miner.go [hunk 2]
```diff
@@ -288,16 +288,10 @@ minerLoop:
...
-			case <-m.stop:
-				stopping := m.stopping
-				m.stop = nil
-				m.stopping = nil
-				close(stopping)
-				return
+			if !m.niceSleep(build.Clock.Until(nextRound)) {
+				continue minerLoop
```
Comment (magik6k) [inline miner/miner.go:292]: I don't think we need this if/continue as this will loop back to the top anyways

File: miner/miner.go (hunk 1 above)
Comment (zgfzgf) [reply miner/miner.go:293]: It might be better to use functions

**Pull request closed without merging**